### PR TITLE
docs - rough serialization and config file updates

### PR DESCRIPTION
**Pull request by jeff-bruemmer**
Lo-fi doc updates to include changes to serialization and the addition of the ability to load Metabase from a config file.

These features are still getting worked out, but wanted to get ahead of this to make sure we're covering everything in docs.

Note https://github.com/metabase/metabase/issues/26388: For this PR, I've included a cheatsheet section for how to convert an environment variable to a config file key, but the idea is we'd be able to generate this info from source code (see the issue).

Feedback at this point: 

- Is everything correct?
- Anything missing?
- Any caveats regarding what you can specify in a config file vs an environment variable?
- How exactly should we communicate serialization v1 to v2? Will the `--legacy` flag make it in?

From there, I'll polish and include some more examples (like serialization output; my understanding is that the YAML output's directory structure is still in flux). Note that these docs should remain as reference documentation; we can follow up with tutorials later on in /learn that step through how exactly you'd use serialization in practice, e.g., to set up a staging-to-production workflow.

### CODE REVIEW

File: docs/configuring-metabase/config-file.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+```
+
+So for example, if you wanted to specify the `MB_EMAIL_FROM_NAME` not in an environment variable, but instead in the `config.yml` file:
+
+```
+version: 1
+config:
+  settings:
+    config-from-file-sync-databases: false
+	email-from-name: Stampy von Mails-a-lot
```
Comment (luizarakaki) [inline docs/configuring-metabase/config-file.md:91]: Is the indentation right?

File: docs/configuring-metabase/config-file.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+## Referring to environment variables in the `config.yml`
+
+Environment variables can be specified with `{% raw %}{{ template-tags }}{% endraw %}` like `{% raw %}{{ env POSTGRES_TEST_DATA_PASSWORD }}{% endraw %}` or `{% raw %}[[options {{template-tags}}]]{% endraw %}`.
+
+Metabase doesn't support recursive expansion, so if one of your environment variables references _another_ environment variable, you're going to have a bad time.
+
+## Disable initial database sync
+
+To disable the initial database sync, you can add `config-from-file-sync-database` to the `settings` list and set the value to `false`. The setting `config-from-file-sync-database` must come _before_ the databases list, like so:
```
Comment (luizarakaki) [inline docs/configuring-metabase/config-file.md:55]: I think it is worth pointing out why users would want to do this.

This only makes sense when they expect to load the data model from a serialized instance and disable the scheduler.

File: docs/installation-and-operation/serialization.md [hunk 3]
```diff
@@ -4,63 +4,126 @@ redirect_from:
...
 
-To help you out in situations like this, Metabase has a serialization feature which lets you create a snapshot, called a dump, of the contents of a Metabase instance that can then be loaded into another instance. This lets you do things like create a set of dashboards and charts in one Metabase instance and then easily copy those items to a number of other Metabase instances. You could also use this feature to enable a staging-to-production workflow for important dashboards or reports by dumping from a staging instance of Metabase and then loading that dump into your production instance(s). You can even put the dump files into version control and audit changes to them, as the YAML files contained within the dump are pretty readable.
+Serialization lets you do things like:
 
-If you're looking to do a one-time migration from H2 to MySQL/Postgres, then use the [migration guide instead](./migrating-from-h2.md).
+- **Snapshot your Metabase**. Serialization creates "dumps" of one Metabase, which you can load into another Metabase.
```
Comment (luizarakaki) [inline docs/installation-and-operation/serialization.md:15]: I think it is worth mentioning that Serialization is **not** a backup mechanism. If they want to backup their instance, there are other ways of doing it.

**Review by luizarakaki — APPROVED** [review]

**Comment by luizarakaki** [comment]
@jeff-bruemmer We are working on some last-minute changes to the serialization structure and it is not clear when we will release it. There is a reasonable chance that it won't make 45.0. I hope we can get it shipped in 45.0 or 45.1.

File: docs/installation-and-operation/serialization.md [hunk 4]
```diff
@@ -4,63 +4,126 @@ redirect_from:
...
+
+```
+java -jar metabase.jar dump [dump_name] --data-model-only true
+```
+
+### `--collections`
+
+- Default: `null` (which dumps all collections)
+
+You can serialize an individual collection, or collections, by using the `--collections` flag in the dump command. To include multiple collections, separate collections with a comma:
```
[the remark this answers is not in the document]
Comment (bshepherdson) [reply docs/installation-and-operation/serialization.md:93]: That's not true. Subcollections will be included if they specify a top-level collection with this flag.

**Review by bshepherdson — APPROVED** [review]
`--legacy` currently doesn't exist, it was to be added by my PR to switch the default to v2, #26206.
Since we're not prepared to switch the default yet, it would perhaps be wise to add `--legacy` in 45 anyway and let users start specifying it if they want.

File: docs/configuring-metabase/config-file.md [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+      last_name: Era
+      password: 2cans3cans4cans
+      email: cam@example.com
+  databases:
+    - name: test-data (Postgres)
+      engine: postgres
+      details:
+        host: localhost
+        port: 5432
+        password: {{ env POSTGRES_TEST_DATA_PASSWORD }}
```
Comment (luizarakaki) [inline docs/configuring-metabase/config-file.md:37]: This requires `"` around like:

`"{{ env POSTGRES_TEST_DATA_PASSWORD }}"

**Comment by jeff-bruemmer** [comment]
Given that we're delaying the release of the new serialization features, I'm only including the docs for the new configuration file feature for paid plans.

**Pull request merged**